source/Test.cpp

```diff
@@ -560,13 +560,13 @@ void Test::Fail(const TestContext &context, const PlayerInfo &player, const stri
 	// Future versions of the test-framework could also print all conditions that are used in the test.
 	string conditions = "";
 	const string TEST_PREFIX = "test: ";
-	auto it = player.Conditions().lower_bound(TEST_PREFIX);
-	for( ; it != player.Conditions().end() && !it->first.compare(0, TEST_PREFIX.length(), TEST_PREFIX); ++it)
+	auto it = player.Conditions().PrimariesLowerBound(TEST_PREFIX);
+	for( ; it != player.Conditions().PrimariesEnd() && !it->first.compare(0, TEST_PREFIX.length(), TEST_PREFIX); ++it)
 		conditions += "Condition: \"" + it->first + "\" = " + to_string(it->second) + "\n";
 
 	if(!conditions.empty())
 		Files::LogError(conditions);
-	else if(player.Conditions().empty())
+	else if(player.Conditions().PrimariesBegin() == player.Conditions().PrimariesEnd())
```

Comment:
You already have an empty() function defined, does this not check for the same thing?
Btw does it only checks for primaries? Probably should be clear in the function's name.

Reply:
The `ConditionsStore::empty()` function does not have a well-defined meaning when we are working with derived conditions. I removed this function.